src/serialbox-fortran/utils_ppser.f90

```diff
@@ -75,17 +76,17 @@ SUBROUTINE ppser_initialize(directory, prefix, mode, prefix_ref, mpi_rank, rprec
   IF ( .NOT. ppser_initialized ) THEN
     IF ( PRESENT(mpi_rank) ) THEN
       WRITE(suffix, '(A5,I0)') "_rank", mpi_rank
-      CALL fs_create_serializer(directory, TRIM(prefix)//TRIM(suffix), 'w', ppser_serializer)
+      CALL fs_create_serializer(directory, TRIM(prefix)//TRIM(suffix), 'w', ppser_serializer, opt_archive)
```

Comment:
Should use a conditional with `PRESENT` otherwise execution will failed when `opt_archive` is not passed.